lib/manager/terraform/extract.spec.ts

```diff
@@ -1,32 +1,63 @@
-import { loadFixture } from '../../../test/util';
-import { extractPackageFile } from './extract';
+import { join } from 'upath';
+import { fs, loadFixture } from '../../../test/util';
+import { setGlobalConfig } from '../../config/global';
+import type { RepoGlobalConfig } from '../../config/types';
+import { extractPackageFile } from '.';
 
 const tf1 = loadFixture('1.tf');
 const tf2 = `module "relative" {
   source = "../../modules/fe"
 }
 `;
 const helm = loadFixture('helm.tf');
+const lockedVersion = loadFixture('lockedVersion.tf');
+const lockedVersionLockfile = loadFixture('rangeStrategy.hcl');
+
+const adminConfig: RepoGlobalConfig = {
+  // `join` fixes Windows CI
+  localDir: join('/tmp/github/some/repo'),
+  cacheDir: join('/tmp/cache'),
+};
+
+// auto-mock fs
+jest.mock('../../util/fs');
 
 describe('manager/terraform/extract', () => {
+  beforeEach(() => {
+    setGlobalConfig(adminConfig);
+  });
   describe('extractPackageFile()', () => {
-    it('returns null for empty', () => {
-      expect(extractPackageFile('nothing here')).toBeNull();
+    it('returns null for empty', async () => {
+      expect(await extractPackageFile('nothing here', '1.tf', {})).toBeNull();
     });
-    it('extracts', () => {
-      const res = extractPackageFile(tf1);
+    it('extracts', async () => {
+      const res = await extractPackageFile(tf1, '1.tf', {});
       expect(res).toMatchSnapshot();
       expect(res.deps).toHaveLength(46);
       expect(res.deps.filter((dep) => dep.skipReason)).toHaveLength(8);
     });
-    it('returns null if only local deps', () => {
-      expect(extractPackageFile(tf2)).toBeNull();
+    it('returns null if only local deps', async () => {
+      expect(await extractPackageFile(tf2, '2.tf', {})).toBeNull();
     });
-    it('extract helm releases', () => {
-      const res = extractPackageFile(helm);
+    it('extract helm releases', async () => {
+      const res = await extractPackageFile(helm, 'helm.tf', {});
       expect(res).toMatchSnapshot();
       expect(res.deps).toHaveLength(6);
       expect(res.deps.filter((dep) => dep.skipReason)).toHaveLength(2);
     });
+
+    it('update lockfile constraints with range strategy update-lockfile', async () => {
+      fs.readLocalFile.mockResolvedValueOnce(lockedVersionLockfile as any);
```

Comment:
```suggestion
      fs.readLocalFile.mockResolvedValueOnce(lockedVersionLockfile);
```

I think those `any` casts are no longer required